iothub_client/src/iothub_client_core.c

```diff
@@ -2486,6 +2486,42 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_UploadMultipleBlocksToBlobAsync(IOTHUB_CLI
     return result;
 }
 
+IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE IoTHubClientCore_CreateUploadContext(IOTHUB_CLIENT_CORE_HANDLE iotHubClientHandle, const char* destinationFileName)
+{
+    IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE result;
+
+    if (
+        (iotHubClientHandle == NULL) ||
+        (destinationFileName == NULL))
+    {
+        LogError("invalid parameters iotHubClientHandle = %p , destinationFileName = %p",
+            iotHubClientHandle,
+            destinationFileName
+        );
+        result = NULL;
+    }
+    else
+    {
+        result = IoTHubClientCore_LL_CreateUploadContext(iotHubClientHandle->IoTHubClientLLHandle, destinationFileName);
+    }
+
+    return result;
+}
+
+IOTHUB_CLIENT_RESULT IoTHubClientCore_UploadBlockToBlob(IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE uploadContextHandle, uint32_t blockNumber, const uint8_t* dataPtr, size_t dataSize)
```

Comment:
I recommend adding another function that allows the application to extract Blob information such as URI/SAS to be used with an external Blob SDK.

Reply:
Yes, as we talked in person. The API has been updated to reflect that.